shell/platform/linux/public/flutter_linux/fl_engine.h

```diff
@@ -23,6 +24,99 @@ G_DECLARE_FINAL_TYPE(FlEngine, fl_engine, FL, ENGINE, GObject)
  * #FlEngine is an object that contains a running Flutter engine.
  */
 
+/**
+ * FlEnginePlatformMessageHandle:
+ *
+ * A handle used to respond to platform messages.
+ */
+typedef struct _FlEnginePlatformMessageHandle FlEnginePlatformMessageHandle;
+
+/**
+ * FlEnginePlatformMessageCallback:
+ * @engine: a #FlEngine
+ * @channel: channel message received on.
+ * @message: message content received from Dart
+ * @handle: a handle to respond to the message with
+ * @user_data: (closure): data provided when registering this callback
+ *
+ * Function called when platform messages are received.
+ */
+typedef void (*FlEnginePlatformMessageCallback)(
+    FlEngine* engine,
+    const gchar* channel,
+    GBytes* message,
+    FlEnginePlatformMessageHandle* handle,
+    gpointer user_data);
+
+/**
+ * fl_engine_set_platform_message_handler:
+ * @engine: a #FlEngine
+ * @channel: channel to listen on
+ * @callback: function to call when a message is received on this channel
+ * @user_data: (closure): user data to pass to @callback
+ *
+ * Register a callback to handle platform messages. Call
+ * fl_engine_send_platform_message_response() when this message should be
+ * responded to.
```

Comment:
All messages should be responded to; failure to respond is a leak. The comment should make that very clear to help avoid people using it incorrectly.